### DISCUSSION TITLE
OpenFileDialog closes my application

### DISCUSSION THREAD

**Discussion by ilia19911** [AvaloniaUI/Avalonia#6131] [General]
Hi. i am trying build my first application in avalonia with button which opens FileDialog.  Then I click on the button, in axaml InvokeCommandAction is triggered and thereafter program opens ReactiveCommand.  My code looks like this
```csharp
async void RunOpenFile(string parameter)
{
    if (Avalonia.Application.Current.ApplicationLifetime is IClassicDesktopStyleApplicationLifetime desktop)
    {
        OpenFileDialog openFileDialog = new OpenFileDialog
        {
            Title = "Choose Images",
        };
        openFileDialog.Filters.Add(new FileDialogFilter { Name = "Images", Extensions = { "jpg", "png", "tif" } });
        await openFileDialog.ShowAsync( desktop.MainWindow);
    }
} 
```
But my application just closes without any exceptions.

OS: Ubuntu 20.04
FW: NET5
Avalonia version: 0.10.6 

I will be glad for any help

**Comment by Gramli** [comment]
Hi,
If RunOpenFile method is invoked by command, you need to catch exceptions in ReactiveCommand Subscribe like for example:

`RunOpenFileCommand.ThrownExceptions.Subscribe(ex=> { /* show ex, log ex etc ...*/ });`

More info: [Reactive UI Commands](https://www.reactiveui.net/docs/handbook/commands/)

**Comment by kekekeks** [comment]
Is there any console output?
